File: C/Mathematical-Fundamentals/isPrime.c
// C Program to find Prime No.
#include <stdio.h>
#include <math.h>
int main()
{
    int n, i, flag = 0;
    printf("Enter a positive integer: ");
    scanf("%d", &n);
    for (i = 2; i <= sqrt(n); ++i)
    {
        // condition for non-prime
        if (n % i == 0)
        {
            flag = 1;
            break;
        }
    }
    if (n == 1)
    {
        printf("1 is neither prime nor composite.");
    }
    else
    {
        if (flag == 0)
            printf("%d is a prime number.", n);
        else
            printf("%d is not a prime number.", n);
    }
    return 0;
}

File: Python/Mathematical-Fundamentals/isPrime.py
# Python Program to find Prime No.
import math

print("Enter a number to check if it is prime or not: ")
num = int(input())
for i in range(2, int(math.sqrt(num))):
    if num % i == 0:
        print("Not Prime")
        break
    else:
        print("Prime")
